locust/dispatch.py

```diff
@@ -62,13 +61,13 @@ def __init__(self, worker_nodes: "List[WorkerNode]", user_classes: List[Type[Use
         assert len(user_classes) > 0
         assert len(set(self._user_classes)) == len(self._user_classes)
 
-        self._target_user_count = None
+        self._target_user_count: int
```

Comment:
This changes self._target_user_count to be undefined (rather than None), right? Maybe that is ok, but I just want to be clear :)

Reply:
Ah crap... if it would be used before actually being set, it would actually throw `AttributeError`, e.g. it would not be `None`.

I see 3 options:
1. Change them to `Optional[..]`, and then deal with all the typing errors that it might cause if it's used without checking that it's not `None`, e.g. changing this one to `Optional[int]` produces:
```
➜ mypy locust/ --ignore-missing-imports
locust/dispatch.py:147: error: Unsupported operand types for < ("int" and "None")
locust/dispatch.py:147: note: Right operand is of type "Optional[int]"
locust/dispatch.py:157: error: Unsupported operand types for > ("int" and "None")
locust/dispatch.py:157: note: Right operand is of type "Optional[int]"
locust/dispatch.py:262: error: Value of type variable "SupportsRichComparisonT" of "min" cannot be "Optional[int]"
locust/dispatch.py:274: error: Unsupported operand types for >= ("int" and "None")
locust/dispatch.py:274: note: Right operand is of type "Optional[int]"
locust/dispatch.py:284: error: Value of type variable "SupportsRichComparisonT" of "max" cannot be "Optional[int]"
locust/dispatch.py:295: error: Unsupported operand types for <= ("int" and "None")
locust/dispatch.py:295: note: Right operand is of type "Optional[int]"
Found 6 errors in 1 file (checked 61 source files)
```
2. Set an initial value of e.g. `0`
3. Use `setattr` to monkey patch them with an initial value of `None`
4. `# type: ignore`

I'd say that the list is ordered by, my, preference. Where 1 would cause most changes.

However, all the tests pass, so internally locust does not seem to try to use the variables before they have actually been set.

Reply:
Perhaps we could use `--no-strict-optional`? It would simplify things in a lot of places.

Reply:
Or maybe... it seems possible to do `self._target_user_count: int = None`?

Reply:
It could definitely be argued that it is weird to have a None-value in something that is not marked as an Optional, but in this case I think it may make sense, because it will not be None for outside users.

Having None-checks all over the place just to keep the type checker happy (option 1) seems excessive (and options 2-4 feel even worse :)

Reply:
Oh man, I was looking for an option to `mypy` that didn't force marking variables `Optional` for having a `None` value. My bad for missing this one.

The combination that worked as `self._target_user_count: int = None` and `--no-strict-optional` parameter to `mypy`.

Reply:
👍  Lets do `--no-strict-optional` for now, and remove the Optional in declarations (except when it is obvious that it may happen even in normal circumstances and thus need to be handled in user code)